includes/jsdev/udf.js

```diff
@@ -0,0 +1,156 @@
+/**
```

Comment:
The majority of the functions do not include the hints and documentation for each one. Can you please add them JSDoc style please.  Also, JS formatting should be Ortus Coding styles.